gradle/libs.versions.toml

```diff
@@ -1,22 +1,22 @@
 [versions]
-micronaut = "4.0.0-M2"
-micronaut-platform = "4.0.0-M2"
+micronaut = "4.0.0-M3"
+micronaut-platform = "4.0.0-SNAPSHOT"
```

Comment:
Changing back to platform snapshot resolved compilation error and classpath issues with different core versions. I suppose we can use the snapshot until platform has final version.

Reply:
It looked like there were both 4.0.0-M2 and 4.0.0-M3 versions of core. I thought it was coming through test-resources or mongodb, but published local snapshots with only M3 and M2 was still present. Couldn't figure out what was the actual root cause, setting platform to snapshot resolved issue.

Reply:
Hmm, maybe we can try to force the Core with:
```
                if (details.requested.group == 'io.micronaut') {
                    if (libs.versions.micronaut instanceof org.gradle.api.provider.Provider) {
                        details.useVersion libs.versions.micronaut.get()
                    } else {
                        details.useVersion libs.versions.micronaut.asProvider().get()
                    }
                }
 ```

Reply:
I hit the same problem in the oracle cloud module. This may fix it https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/725

/cc @melix

Reply:
The issue was with the Graal module added by the Gradle plugin. For me this workaround worked in the oracle cloud module

https://github.com/micronaut-projects/micronaut-oracle-cloud/pull/549/commits/23fa2a1f51ba6aa7f5e43b037637b51d96a08a19

Reply:
> Hmm, maybe we can try to force the Core with:
> 
> ```
>                 if (details.requested.group == 'io.micronaut') {
>                     if (libs.versions.micronaut instanceof org.gradle.api.provider.Provider) {
>                         details.useVersion libs.versions.micronaut.get()
>                     } else {
>                         details.useVersion libs.versions.micronaut.asProvider().get()
>                     }
>                 }
> ```

Looks like you removed that forced section in this PR https://github.com/micronaut-projects/micronaut-data/pull/2191 
Btw, I tried it earlier and got this error

> Could not find io.micronaut:micronaut-bom:4.0.0-M3

Reply:
Fixed by importing core like this in that doc-example project
```micronautBoms(platform("io.micronaut:micronaut-core-bom:$micronautVersion"))```